src/Features/Blockcore.Features.Consensus/ConsensusFeature.cs

```diff
@@ -56,11 +56,11 @@ public override Task InitializeAsync()
             {
                 // Add witness discovery as a requirement if witness is activated.
                 this.connectionManager.AddDiscoveredNodesRequirement(NetworkPeerServices.NODE_WITNESS);
-
-                // Set witness as a supported service if witness is activated.
-                this.connectionManager.Parameters.Services |= NetworkPeerServices.NODE_WITNESS;
             }
 
+            // Always announce that the node supports WITNESS even if witness is not activated yet.
+            this.connectionManager.Parameters.Services |= NetworkPeerServices.NODE_WITNESS;
```

Comment:
I think its correct to only put that on service , it means when segwit is not active the peer will still connect to older nodes that support segwit but dont broadcast that